v03_pipeline/lib/reference_data/reference_dataset.py

```diff
@@ -0,0 +1,106 @@
+from enum import Enum
+
+from v03_pipeline.lib.model import AccessControl, DatasetType, Env, ReferenceGenome
+from v03_pipeline.lib.reference_data.cadd import load_cadd_ht_from_raw_dataset
+from v03_pipeline.lib.reference_data.hgmd import download_and_import_hgmd_vcf
+
+
+class ReferenceDataset(str, Enum):
+    cadd = 'cadd'
+    hgmd = 'hgmd'
+
+    @classmethod
+    def for_reference_genome_dataset_type(
+        cls,
+        reference_genome: ReferenceGenome,
+        dataset_type: DatasetType,
+    ) -> list['ReferenceDataset']:
+        reference_datasets = {
+            (ReferenceGenome.GRCh37, DatasetType.SNV_INDEL): [
+                ReferenceDataset.cadd,
+                ReferenceDataset.hgmd,
+            ],
+            (ReferenceGenome.GRCh38, DatasetType.SNV_INDEL): [
+                ReferenceDataset.cadd,
+                ReferenceDataset.hgmd,
+            ],
+        }.get((reference_genome, dataset_type), [])
+        if not Env.ACCESS_PRIVATE_REFERENCE_DATASETS:
+            return [
+                rd
+                for rd in reference_datasets
+                if rd.access_control == AccessControl.PUBLIC
+            ]
+        return reference_datasets
+
+    @property
+    def access_control(self) -> AccessControl:
+        if self == ReferenceDataset.hgmd:
+            return AccessControl.PRIVATE
+        return AccessControl.PUBLIC
+
+    def version(self, reference_genome: ReferenceGenome):
+        return {
+            (ReferenceDataset.cadd, ReferenceGenome.GRCh37): 'v1.7',
+            (ReferenceDataset.cadd, ReferenceGenome.GRCh38): 'v1.7',
+            (ReferenceDataset.hgmd, ReferenceGenome.GRCh37): 'HGMD_Pro_2023',
+            (ReferenceDataset.hgmd, ReferenceGenome.GRCh38): 'HGMD_Pro_2023',
+        }[self, reference_genome]
+
+    def raw_dataset_path(self, reference_genome: ReferenceGenome):
+        return {
+            (ReferenceDataset.cadd, ReferenceGenome.GRCh37): {
+                'snv': 'https://krishna.gs.washington.edu/download/CADD/v1.7/GRCh37/whole_genome_SNVs.tsv.gz',
+                'indel': 'https://krishna.gs.washington.edu/download/CADD/v1.7/GRCh38/gnomad.genomes.r4.0.indel.tsv.gz',
+            },
+            (ReferenceDataset.cadd, ReferenceGenome.GRCh38): {
+                'snv': 'https://krishna.gs.washington.edu/download/CADD/v1.7/GRCh38/whole_genome_SNVs.tsv.gz',
```

Comment:
Are there any other cases where there's multiple files?  I not maybe just pass a list here and `union` upstream?

Reply:
I _think_ this is the only case with multiple files